### ISSUE TITLE
<tab> in label-mode should un-highlight skipped matches

### ISSUE THREAD

**Issue by justinmk** [justinmk/vim-sneak#125]

**Comment by justinmk** [comment]
> much nicer where the cursor never moves from its initial position (but still un-highlights the skipped matches)

If the un-highlight works then the cursor as a visual-aid is less important. But there is still a cost, which is that if sneak is dismissed, then the cursor placement is different than what was indicated. A key feature is that you can dismiss sneak at any time painlessly.

> Not sure yet how much of a headache that would be to implement in vim-sneak

Yes, a complicated implementation would be a dealbreaker. A simple implementation is worth considering.

> TBH I've settled for quite a small set of labels ... finding the `<Tab>` (or leap `<Space>`) preferable to a hard-to-press label.

Yes, it's an intentional decision of vim-sneak to not do the "extended labels" thing, it presents the user with a "puzzle" which raises the question of why not use `/` or other navigation mechanisms instead. `<Tab>` is the escape-hatch for rare cases or where you might want dot-repeat.

**Comment by ggandor** [comment]
> it's an intentional decision of vim-sneak to not do the "extended labels" thing, it presents the user with a "puzzle" which raises the question of why not use `/` or other navigation mechanisms instead. <Tab> is the escape-hatch for rare cases

I don't understand this argument at all. An "extended label set" simply means using more labels, where appropriate. It actually decreases the _steps_ needed for a jump on average. Where is the puzzle in that?

> If the un-highlight works then the cursor as a visual-aid is less important. But there is still a cost, which is that if sneak is dismissed, then the cursor placement is different than what was indicated.

According to this proposal, the cursor does actually stay in place, `<Tab>` just switches highlights. My educated guess is that you have a gut reaction against introducing state (which set of labels are "active"), but that doesn't complicate the implementation that much compared to the gain.

> why not use / or other navigation mechanisms instead

Arguments along this line sound a tiny bit like a thought-terminating cliché to me (cf. "a text editor should not have a terminal...") :) Because you don't want to compose multiple atomic motion commands, or press `n` 10 times, _or_ play Scrabble with `/` (if that's possible at all). `/` and other methods are suboptimal for quickly moving the cursor to arbitrary places in the window (reliably).

**Comment by justinmk** [comment]
> Still I don't get it why we need to _intentionally_ limit the efficiency of the escape hatch, when it could decrease the steps

"could" is the operative word, it means that a 1-char target *may* become a 2-char target, which means you never know, even if your actual desired target is within the first few matches.

> Sneak vs `/` is _already_ a "puzzle".

All of the constraints in Sneak are intended to make it predictable, it doesn't change behavior depending on number of targets.

**Comment by justinmk** [comment]
> The first set should _always_ be "safe", but once the user has used `<Tab>`, there is no reason to use a limited set

Oh, yes that makes sense. Only question then is how complicated is the logic.

Mainly I think `<Tab>` is not something that needs to be optimized because it is inherently a UI failure.

**Comment by tuurep** [comment]
Yeah ggandor I see your point, thanks for persisting on it though seems fairly painful 😄 I think, finally I agree with you.
